Add GitHub Action

ermshiperete opened the pull request.

ermshiperete committed:
Add GitHub Action

ermshiperete committed:
Move common properties to Directory.Build.props

ermshiperete committed:
Update dependencies

ermshiperete committed:
Replace MarkdownDeep with Markdig

ermshiperete committed:
Target .NET 4.7.2 instead of 4.6.1

+semver:minor

ermshiperete committed:
Remove unnecessary depdendencies on Microsoft.Build.*

ermshiperete committed:
Set environment variables for Visual Studio

This is necessary to be able to compile the unmanaged ForceCrash
test lib. This change also directly calls compile on Windows
without going through a .vcxproj file.

tombogle commented on SIL.BuildTasks.AWS/SIL.BuildTasks.AWS.csproj:
```diff
@@ -1,52 +1,22 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net461</TargetFrameworks>
+    <TargetFrameworks>net472</TargetFrameworks>
```
Am I correct in assuming that this will not prevent projects that target 4.6.1 from using SIL.BuildTasks?

ermshiperete replied:
Yes, that's correct. Since this runs in the context of `msbuild`/`dotnet build` it is independent of the target framework. I guess you might get into trouble if you're trying to build on a very old machine that doesn't have .NET Framework >= 4.7 available...

ermshiperete committed:
Refactor NUnit tests

This allows to run them with `dotnet test`. Also rename
existing tests for NUnit to NUnitIntegrationTests and create
new unit tests which allow for easier debugging.

ermshiperete committed:
Update readme

Also add build badge.

tombogle commented on SIL.BuildTasks.Tests/UnitTestTasks/NUnitIntegrationTests.cs:
```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2016-2022 SIL International
```
Not sure it matters much, but if this is a new file, its copyright probably can't go back to 2016.

ermshiperete replied:
It's basically a copy of the old `NUnitTests.cs` file because the tests were really integration tests.
